ra/registration-authority.go

```diff
@@ -680,7 +680,17 @@ func (ra *RegistrationAuthorityImpl) checkLimits(names []string, regID int64) er
 	if limits.CertificatesPerName.Enabled() {
 		err := ra.checkCertificatesPerNameLimit(names, limits.CertificatesPerName, regID)
 		if err != nil {
-			return err
+			if _, ok := err.(core.RateLimitedError); !ok {
```

Comment:
I think this whole chunk should probably be inside checkCertificatesPerNameLimit. Then you can return early if err is nil and outdent the rest.